src/connector.js

```diff
@@ -901,16 +912,12 @@ function updateQuerySummaryState( newState ) {
 		querySummaryElement.textContent = "";
 		if ( querySummaryState.total > 0 ) {
 			let numberOfResults = querySummaryState.total.toLocaleString( params.lang );
-			// Create the <h2> element
-			const hTwoAnchor = document.createElement("h2");
 			// Generate the text content
-			const querySummaryText = ((querySummaryState.query !== "" && !params.isAdvancedSearch) ? querySummaryTemplateHTML : noQuerySummaryTemplateHTML)
-				.replace('%[numberOfResults]', numberOfResults)
-				.replace('%[query]', querySummaryState.query)
-				.replace('%[queryDurationInSeconds]', querySummaryState.durationInSeconds.toLocaleString(params.lang));
-			hTwoAnchor.textContent = querySummaryText;
-			querySummaryElement.innerHTML = "";
-			querySummaryElement.appendChild(hTwoAnchor);
+			const querySummaryHTML = ( ( querySummaryState.query !== "" && !params.isAdvancedSearch ) ? querySummaryTemplateHTML : noQuerySummaryTemplateHTML )
+				.replace( '%[numberOfResults]', numberOfResults )
+				.replace( '%[query]', querySummaryState.query )
+				.replace( '%[queryDurationInSeconds]', querySummaryState.durationInSeconds.toLocaleString( params.lang ) );
+			querySummaryElement.innerHTML = querySummaryHTML;
```

Comment:
@bblaisATcoveo This approach reverts the implemented security fix. We need to look at a different solution.

Reply:
@GormFrank code adjusted below